### DISCUSSION TITLE
Hide 'View query' option from dashboard

### DISCUSSION THREAD

**Discussion by Narendra678** [apache/superset#18807] [Q&A / Help]
Hi Team,

I need to hide 'View query' option from Dashboard when click on three dots.

![image](https://user-images.githubusercontent.com/88739186/154541800-f6686110-a206-4720-9cf0-8e7ea85bc2e7.png)

Thanks in advance.

Regards,
Naren

**Comment by srinify** [comment]
Hi Naren, moving to Discussions as this isn't a bug or feature request!

**Comment by KarthikGhorpade** [comment]
Hi @Narendra678, were you able to find a solution for this?

**Comment by kamalkeshavani-aiinside** [comment]
I think users without Explore Chart permission, will not see this option in dashboard

**Comment by KarthikGhorpade** [comment]
Hi @kamalkeshavani-aiinside, in my case, we would like the users to be able to 'Export CSV' / 'Download -> CSV' but to restrict them from seeing / viewing the query. Are you aware of any permissions that enable us to achieve this?

**Reply by kamalkeshavani-aiinside** [reply]
I am using a custom role with below permissions and `can csv on Superset` to allow user to just view dashboards and download csv.
<img width="887" alt="image" src="https://user-images.githubusercontent.com/74634977/183563060-20a8b158-71fd-4c87-bfc1-d1fbee2a701f.png">

**Reply by KarthikGhorpade** [reply]
I will check if this works for the use-case. Thanks a lot!

**Comment by sfirke** [comment]
Remove the `explore on Superset` privilege and that will remove the View Query option.  A side effect of that will be that the role can no longer view embedded standalone charts.  That same permission controls both.

My workaround for this blocking standalone charts in an iframe is to make a single-chart dashboard and then embed that.

**Reply by vedfordev** [reply]
It's working 👍

**Reply by KarthikGhorpade** [reply]
@sfirke I had not noticed your answer until today. Thanks a lot! :)

**Reply by pinarsari** [reply]
Hi, does it works? Can you see all dashboards/charts without any problem after removing "explore on Superset" option?
